src/wms/src/test/java/org/geoserver/wms/featureinfo/GetFeatureInfoJSONTest.java

```diff
@@ -361,4 +372,130 @@ public void testDateTimeFormattingDisabled() throws Exception {
             TimeZone.setDefault(defaultTimeZone);
         }
     }
+
+    /** Tests json output mediated by a free marker template */
+    @Test
+    public void testJSONFreeMarkerTemplate() throws Exception {
```

Comment:
Good start for testing, but insufficient. Should check the multi-collection case (try with a layer group) and the mix of layers with and without templates.

Reply:
fixed